modules/shells/posix.nix

```diff
@@ -46,22 +46,21 @@ with lib; {
         mapAttrsToList (n: v: ''export ${n}="${concatStringsSep ":" v}"'') allValuesLists;
     in
       concatStringsSep "\n" exportVariables;
-    activateScript = builtins.toFile "activate" ''
-      ${exportedEnvVars}
+      activateScript = pkgs.writeTextFile {
+        name =  "activate";
+        executable = true;
+        destination = "/activate";
+        text = ''
+          ${exportedEnvVars}
 
-      ${stringAliases}
+          ${stringAliases}
 
-      ${config.shell.hook}
-    '';
+          ${config.shell.hook}
+        '';
+      };
   in {
-    toplevel = pkgs.stdenvNoCC.mkDerivation {
-      name = "floxEnv";
-      buildCommand = ''
-        mkdir $out
-        ln -s ${config.system.path} $out/sw
-        ln -s ${config.newCatalogPath} $out/catalog.json
-        ln -s ${activateScript} $out/activate
-      '';
+    toplevel = pkgs.callPackage ../../lib/mkEnv.nix {} {
```

Comment:
Want to make sure I'm understanding the direction you want this to go...is your goal more to support `nix develop` or to avoid re-implementing the activation script?
- If it's the former, I think we can strip some more stuff out of `mkEnv` (doesn't have to be right now, just at some point). In that case `mkEnv` would become more of a `wrapThesePackagesAndActivationScriptSoTheySupportDevelop`
- If it's the latter, we could drop generating the activation script in this file, and rely on `mkEnv` to do the generation. Then it's `mkEnv = {aliases, env_vars, hooks}`

Reply:
Discussed live